core/src/test/java/org/hisp/dhis/android/core/configuration/internal/DatabaseNameGeneratorShould.kt

```diff
@@ -38,42 +38,121 @@ class DatabaseNameGeneratorShould {
     private val urlHttp = "http://play.dhis2.org/android-current"
     private val username = "user23"
     private val generator = DatabaseNameGenerator()
-    private val expectedEncr = "play-dhis2-org-android-current_user23_encrypted.db"
-    private val expectedUnen = "play-dhis2-org-android-current_user23_unencrypted.db"
 
     @Test
-    fun return_name_with_server_and_username_with_valid_characters_for_encrypted_db() {
+    fun return_name_with_server_username_hash_and_encryption_for_encrypted_db() {
         val name = generator.getDatabaseName(url, username, true)
-        assertThat(name).isEqualTo(expectedEncr)
+
+        // Should have format: <readable>_<username>_<hash>_encrypted.db
+        assertThat(name).matches("play-dhis2-org-android-current_user23_[0-9a-f]{8}_encrypted\\.db")
+        assertThat(name).contains("_user23_")
+        assertThat(name).endsWith("_encrypted.db")
     }
 
     @Test
-    fun return_name_with_server_and_username_with_valid_characters_for_unencrypted_db() {
+    fun return_name_with_server_username_hash_and_encryption_for_unencrypted_db() {
         val name = generator.getDatabaseName(url, username, false)
-        assertThat(name).isEqualTo(expectedUnen)
+
+        assertThat(name).matches("play-dhis2-org-android-current_user23_[0-9a-f]{8}_unencrypted\\.db")
+        assertThat(name).endsWith("_unencrypted.db")
+    }
+
+    @Test
+    fun return_same_name_for_http_and_https() {
+        val nameHttps = generator.getDatabaseName(url, username, true)
+        val nameHttp = generator.getDatabaseName(urlHttp, username, true)
+
+        assertThat(nameHttps).isEqualTo(nameHttp)
```

Comment:
Maybe we can discuss this point, I am not sure if instances with different protocols should be considered the same.